### PR TITLE
[expo-go] remove InstallReferrerReceiver

### PR DESCRIPTION
**Pull request by vonovak**
# Why

fixes `com.google.android.gms.common.util.VisibleForTesting` build time error

`InstallReferrerReceiver` inherits from `CampaignTrackingReceiver` which used `VisibleForTesting`

The build failures are expected and fixed up the stack

```
> Task :expoview:compileDebugKotlin FAILED
e: file:///home/expo/workingdir/build/apps/expo-go/android/expoview/src/main/java/host/exp/exponent/fcm/ExpoFirebaseMessagingDelegate.kt:56:3 'createNotificationRequest' overrides nothing
e: file:///home/expo/workingdir/build/apps/expo-go/android/expoview/src/main/java/host/exp/exponent/notifications/ScopedCategoryAwareNotificationBuilder.kt:23:9 Type mismatch: inferred type is String? but String was expected
...
```

# How

remove `InstallReferrerReceiver`

# Test Plan

no build error related to VisibleForTesting

# Checklist

- [ ] Documentation is up to date to reflect these changes (eg: https://docs.expo.dev and README.md).
- [ ] Conforms with the [Documentation Writing Style Guide](https://github.com/expo/expo/blob/main/guides/Expo%20Documentation%20Writing%20Style%20Guide.md)
- [ ] This diff will work correctly for `npx expo prebuild` & EAS Build (eg: updated a module plugin).

### CODE REVIEW

**Comment by vonovak** [comment]
> [!WARNING]
> <b>This pull request is not mergeable via GitHub because a downstack PR is open. Once all requirements are satisfied, merge this PR as a stack <a href="https://app.graphite.dev/github/pr/expo/expo/31735?utm_source=stack-comment-downstack-mergeability-warning" >on Graphite</a>.</b>
> <a href="https://graphite.dev/docs/merge-pull-requests">Learn more</a>

* **#31737** <a href="https://app.graphite.dev/github/pr/expo/expo/31737?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#31736** <a href="https://app.graphite.dev/github/pr/expo/expo/31736?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#31735** <a href="https://app.graphite.dev/github/pr/expo/expo/31735?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈
* **#31734** <a href="https://app.graphite.dev/github/pr/expo/expo/31734?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* `main`

This stack of pull requests is managed by Graphite. <a href="https://stacking.dev/?utm_source=stack-comment">Learn more about stacking.</a>
<h2></h2>

 Join @vonovak and the rest of your teammates on <a href="https://graphite.dev?utm-source=stack-comment">Graphite <b>Graphite</b></a>

**Review by gabrieldonadel — APPROVED** [review]
LGTM

**Review by Kudo — APPROVED** [review]

File: apps/expo-go/android/expoview/src/main/java/host/exp/exponent/kernel/Kernel.kt [hunk 1]
```diff
@@ -341,25 +340,9 @@ class Kernel : KernelInterface() {
...
-      try {
-        bundle.putBundle("exp", BundleJSONConverter.convertToBundle(exponentProps))
-      } catch (e: JSONException) {
-        throw Error("JSONObject failed to be converted to Bundle", e)
-      }
-      return bundle
-    }
+  private val kernelLaunchOptions = Bundle().apply {
+    putBundle("exp", Bundle())
+  }
```
[the remark this answers is not in the document]
Comment (vonovak) [reply apps/expo-go/android/expoview/src/main/java/host/exp/exponent/kernel/Kernel.kt]: thanks, that's a bit nicer. Though I'm not a fan of the fact that Kotlin always has 10 ways of achieving the same thing 🤷‍♂️

**Comment by vonovak** [comment]
### Merge activity

* **Sep 28, 5:18 AM EDT**: @vonovak started a stack merge that includes this pull request via [Graphite](https://app.graphite.dev/github/pr/expo/expo/31735).

**Pull request merged**